### PR TITLE
(SIMP-5084) Use OS keys in simp_filesystem.repo

### PR DESCRIPTION
**Pull request by op-ct**
Before this commit, the OS GPG key in `simp_filesystem.repo` was
hardcoded to the corresponding major release of CentOS, regardless of
the OS being used.

On RHEL builds of SIMP, this would result in the ommission of the GPG
key file `RPM-GPG-KEY-redhat-release`, causing many packages to fail to
install during Puppet runs very early in the SIMP master deployment
process.

This patch fixes the issues by adding logic to the ISO's `auto.cfg` file
to include an OS-specific GPG key file (for the `[flocal-${arch}]` and
`[frhbase]` repos) while constructing `simp_filesystem.repo`.

SIMP-5084 #close

### CODE REVIEW

File: Changelog.rst [hunk 1]
```diff
@@ -609,6 +609,12 @@ pupmod-simp-useradd
 * Set the min and max ``UID`` and ``GID`` based on what is in ``login.defs``
   and default to something sensible for the platform.
 
+
+simp-core
```
Comment (lnemsick-simp) [inline Changelog.rst]: Have we ever noted the changes to simp-core in the changelog?  If so, there are a whole bunch of changes for this release related to EFI that need to be added.

File: build/distributions/CentOS/6/x86_64/DVD/ks/dvd/auto.cfg [hunk 2]
```diff
@@ -255,18 +275,18 @@ gpgkey=file:///var/www/yum/SIMP/GPGKEYS/RPM-GPG-KEY-puppet
     file:///var/www/yum/SIMP/GPGKEYS/RPM-GPG-KEY-grafana
     file:///var/www/yum/SIMP/GPGKEYS/RPM-GPG-KEY-PGDG-96
     file:///var/www/yum/SIMP/GPGKEYS/RPM-GPG-KEY-EPEL-$majrhversion
-    file:///var/www/yum/SIMP/GPGKEYS/RPM-GPG-KEY-CentOS-$majrhversion
-    $extrakey
+$flocal_gpg_file_lines
```
Comment (lnemsick-simp) [inline build/distributions/CentOS/6/x86_64/DVD/ks/dvd/auto.cfg]: To absolutely ensure the spacing is correct (which yum requires), I would add all the keys into the array (grafana, PDDG, EPEL...)

File: build/distributions/CentOS/6/x86_64/DVD/ks/dvd/auto.cfg [hunk 3]
```diff
@@ -228,20 +228,40 @@ rhversion=`facter operatingsystemrelease`
...
+    flocal_gpg_files+=("$frhbase_gpg_file")
+  fi
 
-  extrakey=''
   if [ -e /var/www/yum/SIMP-Dev/GPGKEYS/RPM-GPG-KEY-SIMP-Dev ]; then
-    extrakey='file:///var/www/yum/SIMP-Dev/GPGKEYS/RPM-GPG-KEY-SIMP-Dev'
+    flocal_gpg_files+=('file:///var/www/yum/SIMP-Dev/GPGKEYS/RPM-GPG-KEY-SIMP-Dev')
   fi
 
+  flocal_gpg_file_lines="$(IFS=$'\n'; echo "${flocal_gpg_files[*]}" | sed -e 's/^\([a-z]\)/    \1/g' -e '/^[[:space:]]*$/d')"
```
Comment (lnemsick-simp) [inline build/distributions/CentOS/6/x86_64/DVD/ks/dvd/auto.cfg]: I hate bash.  This is ugly.

File: src/assets/simp/build/simp.spec [hunk 4]
```diff
@@ -221,6 +221,9 @@ fi
 # Post uninstall stuff
 
 %changelog
+* Mon Aug 27 2018 Chris Tessmer <chris.tessmer@onyxpoint.com> - 6.2.0-0
```
Comment (lnemsick-simp) [inline src/assets/simp/build/simp.spec]: This change is in the ISO, not in the simp RPM.  Please remove this comment

**Comment by trevor-vaughan** [comment]
This did not appear to fix the issue, working on debugging.

**Pull request merged**